src/contexts/course-mode/application/use-cases/add-user-message-to-chat-concrete.ts

```diff
@@ -0,0 +1,22 @@
+import { IdProvider } from "@/shared/application/ports/out/id-provider";
+import { AddUserMessageToChat } from "../ports/in/add-user-message-to-chat";
+import { Conversation } from "@/shared/entities/conversation";
+import { Message, Role } from "@/shared/application/ports/out/llm-provider";
+
+export class AddUserMessageToChatConcrete implements AddUserMessageToChat {
```

Comment:
we could call it `AddUserMessageToChatUseCase` wdyt? the file name could be simply add-user-to-chat.ts too, as the port is in different dir, personally the duplicated name doesn't bother me